petastorm/spark/spark_dataset_converter.py

```diff
@@ -478,7 +479,8 @@ def _cache_df_or_retrieve_cache_data_url(df, parent_cache_dir_url,
             if meta.row_group_size == parquet_row_group_size_bytes and \
                     meta.compression_codec == compression_codec and \
                     meta.df_plan.sameResult(df_plan) and \
-                    meta.dtype == dtype:
+                    meta.dtype == dtype and \
+                    meta.parent_cache_dir_url == parent_cache_dir_url:
```

Comment:
Before comparing, let's apply `petastorm.reader.normalize_dataset_url` first.

Reply:
I added the comparison after getting the conf so that all the saved parent_cache_dir is already normalized.